### ISSUE TITLE
Pagination support

### ISSUE THREAD

**Issue by SeyZ** [intercom/intercom-ruby#215]
The following call does not retrieve 5 users, but all users. The page parameters seems to be not supported.

```
intercom.users.find_all(type: 'users', page: 1, per_page: 5, order: :asc).to_a
# --> returns all my users instead of just 5
```

**Comment by luiz-s81** [comment]
This is a pretty much needed feature in the Ruby SDK.

**Comment by olliebennett** [comment]
It's worth noting perhaps that [this doc](https://developers.intercom.io/reference#list-users) indicates that the maximum `per_page` value is `50` (the default). This means that changes to this gem will not (cannot) help those of us wanting to _increase_ the page size.

The issue remains valid for the OP's use-case of _decreasing_ the page size.

**Comment by choran** [comment]
HI, We added support for this feature which is now available in version 3.5.2.
Please take a look and reopen if you have any questions/issues.
And as @olliebennett noted, unfortunately, we cannot increase the page size from the default of 50, you can now set it to you own lower limits as needed.

**Comment by flood4life** [comment]
@choran API documentation states that `page` param defines what page of result set to fetch as first, not last, i.e., `intercom.users.find_all(order: :asc, page: 2)` should just skip first 50 users and return all users after those, and that was correct until 3.5.2. Now, if `page` param is supplied, only a single page is retrieved. Perhaps it's better to distinguish these two params? E.g.,  `intercom.users.find_all(order: :asc, start_page: 2, end_page: 10)` would retrieve users from 51 to 500.
